chia/rpc/wallet_rpc_api.py

```diff
@@ -3120,18 +3120,11 @@ async def create_new_dl(self, request) -> Dict:
         if self.service.wallet_state_manager is None:
             raise ValueError("The wallet service is not currently initialized")
 
-        dl_wallet: DataLayerWallet
-        for _, wallet in self.service.wallet_state_manager.wallets.items():
-            if WalletType(wallet.type()) == WalletType.DATA_LAYER:
-                assert isinstance(wallet, DataLayerWallet)
-                dl_wallet = wallet
-                break
-        else:
+        try:
+            dl_wallet = self.service.wallet_state_manager.get_dl_wallet()
+        except ValueError:
```

Comment:
I think this could be in the lock?  Or am I misunderstanding the existing comments about locking etc.  At which point basically just the whole function could be in a single acquisition of the lock.

Reply:
Kind of yes, for the very first time its called there is a race and but after i guess it doesn't matter because there always will be only DL wallet. But still it seems there might be some cases where we could get "size changed during iteration" issues... However, this PR just retains behaviour while cleaning up some duplicated code now, moving it into the lock means those endpoints will block also for wallet fetching so i think instead of just locking there, we should cache the DL wallet separately in the wallet state manager since it can only be one and just access the cached wallet.